app/assets/javascripts/admin/addon/components/admin-search.gjs

```diff
@@ -0,0 +1,124 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { concat } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import { TrackedObject } from "@ember-compat/tracked-built-ins";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
+import DButton from "discourse/components/d-button";
+import icon from "discourse/helpers/d-icon";
+import discourseDebounce from "discourse/lib/debounce";
+import { INPUT_DELAY } from "discourse/lib/environment";
+import autoFocus from "discourse/modifiers/auto-focus";
+import { i18n } from "discourse-i18n";
+import AdminSearchFilters from "admin/components/admin-search-filters";
+import { RESULT_TYPES } from "admin/services/admin-search-data-source";
+
+export default class AdminSearch extends Component {
+  @service adminSearchDataSource;
+
+  @tracked filter = "";
+  @tracked searchResults = [];
+  @tracked showFilters = false;
+  @tracked loading = false;
+  @tracked
+  typeFilters = new TrackedObject({
+    page: true,
+    setting: true,
+    theme: true,
+    component: true,
+    report: true,
+  });
+
+  constructor() {
+    super(...arguments);
+    this.adminSearchDataSource.buildMap();
+  }
+
+  get visibleTypes() {
+    return Object.keys(this.typeFilters).filter(
+      (type) => this.typeFilters[type]
+    );
+  }
+
+  get showLoadingSpinner() {
+    return !this.adminSearchDataSource.isLoaded || this.loading;
+  }
+
+  @action
+  toggleFilters() {
+    this.showFilters = !this.showFilters;
```

Comment:
Not sure if it's worth using Ember [`#toggleProperty`](https://api.emberjs.com/ember/6.1/classes/Component/methods/toggleProperty?anchor=toggleProperty)?

Reply:
Hmm this doesn't work, I think it's kinda "old ember"